packages/plugin/src/theia.d.ts

```diff
@@ -2629,6 +2629,105 @@ declare module '@theia/plugin' {
         attributes?: { [key: string]: string | null };
     }
 
+    /**
+	 * Represents the dimensions of a terminal.
+	 */
+    export interface TerminalDimensions {
+		/**
+		 * The number of columns in the terminal.
+		 */
+        readonly columns: number;
+
+		/**
+		 * The number of rows in the terminal.
+		 */
+        readonly rows: number;
+    }
+
+    /**
+     * Value-object describing what options a virtual process terminal should use.
+     */
+    export interface ExtensionTerminalOptions {
+		/**
+		 * A human-readable string which will be used to represent the terminal in the UI.
+		 */
+        name: string;
+
+		/**
+		 * An implementation of [Pseudoterminal](#Pseudoterminal) that allows an extension to
+		 * control a terminal.
+		 */
+        pty: Pseudoterminal;
+    }
+
+	/**
+	 * Defines the interface of a terminal pty, enabling extensions to control a terminal.
+	 */
+    interface Pseudoterminal {
+		/**
+		 * An event that when fired will write data to the terminal. Unlike
+		 * [Terminal.sendText](#Terminal.sendText) which sends text to the underlying _process_
+		 * (the pty "slave"), this will write the text to the terminal itself (the pty "master").
+		 */
+        onDidWrite: Event<string>;
+
+		/**
+		 * An event that when fired allows overriding the [dimensions](#Pseudoterminal.setDimensions) of the
+		 * terminal. Note that when set, the overridden dimensions will only take effect when they
+		 * are lower than the actual dimensions of the terminal (ie. there will never be a scroll
+		 * bar). Set to `undefined` for the terminal to go back to the regular dimensions (fit to
+		 * the size of the panel).
+		 */
+        onDidOverrideDimensions?: Event<TerminalDimensions | undefined>;
+
+		/**
+		 * An event that when fired will signal that the pty is closed and dispose of the terminal.
+		 *
+		 * A number can be used to provide an exit code for the terminal. Exit codes must be
+		 * positive and a non-zero exit codes signals failure which shows a notification for a
+		 * regular terminal and allows dependent tasks to proceed when used with the
+		 * `CustomExecution` API.
+		 */
+        onDidClose?: Event<void | number>;
+
+		/**
+		 * Implement to handle when the pty is open and ready to start firing events.
+		 *
+		 * @param initialDimensions The dimensions of the terminal, this will be undefined if the
+		 * terminal panel has not been opened before this is called.
+		 */
+        open(initialDimensions: TerminalDimensions | undefined): void;
+
+		/**
+		 * Implement to handle when the terminal is closed by an act of the user.
+		 */
```

Comment:
something weird with formatting, could you apply auto formatting in Theia or VS Code please?

Reply:
![image](https://user-images.githubusercontent.com/5847142/83267831-5dbec980-a1f7-11ea-8cd5-2291d872868d.png)
Its weird, formatting looks good in my vscode

Reply:
Are you using tabs to indent the jsdoc?

Reply:
Good catch; tabs. That's the problem.

Reply:
Aha, I will fix it :p